starknet_py/tests/e2e/client/client_test.py

```diff
@@ -49,44 +48,35 @@ async def test_get_deploy_transaction(
 
 
 @pytest.mark.asyncio
-async def test_get_declare_transaction(clients, declare_transaction_hash, class_hash):
+async def test_get_declare_transaction(
+    clients, declare_transaction_hash, class_hash, gateway_account_client
+):
     # TODO extend this test to all clients
     gateway_client, _ = clients
 
     transaction = await gateway_client.get_transaction(declare_transaction_hash)
-
-    assert transaction == DeclareTransaction(
-        class_hash=class_hash,
-        sender_address=DEFAULT_DECLARE_SENDER_ADDRESS,
-        hash=declare_transaction_hash,
-        signature=[],
-        max_fee=0,
-        version=0,
-        nonce=None,
+    sender_address = (
+        gateway_account_client.address
+        if transaction.version == 1
+        else DEFAULT_DECLARE_SENDER_ADDRESS
     )
 
+    assert transaction.class_hash == class_hash
+    assert transaction.hash == declare_transaction_hash
+    assert transaction.sender_address == sender_address
+
 
 @pytest.mark.asyncio
 async def test_get_invoke_transaction(
     clients,
     invoke_transaction_hash,
     invoke_transaction_calldata,
-    invoke_transaction_selector,
-    contract_address,
 ):
     for client in clients:
         transaction = await client.get_transaction(invoke_transaction_hash)
 
-        assert transaction == InvokeTransaction(
-            contract_address=contract_address,
-            calldata=invoke_transaction_calldata,
-            entry_point_selector=invoke_transaction_selector,
-            hash=invoke_transaction_hash,
-            signature=[],
-            max_fee=0,
-            version=0,
-            nonce=None,
-        )
+        assert invoke_transaction_calldata[0] in transaction.calldata
```

Comment:
It checks just a single value from calldata, right?

Reply:
Yes, because transaction.calldata has many values and we are interested only in invoke_transaction_calldata[0]

Reply:
Why are we only interested in the first one?

Reply:
because we pass [1234] as constructor argument and transaction.calldata is something like [0, 5, ..., 1234].
So our argument is here, but there are additional values too

Reply:
And we can't explicitly do `assert invoke_transaction_calldata[0] in transaction.calldata`

Reply:
Are we sure constructor calldata will always be at 0th place? Perhaps `any(1234 in data for data in calldata)` would be a safer check?